src/cpp/src/module_genai/module_base.hpp

```diff
@@ -63,8 +64,10 @@ class IBaseModule {
 public:
     ~IBaseModule() = default;
     using PTR = std::shared_ptr<IBaseModule>;
+    using WEAK_PTR = std::weak_ptr<IBaseModule>;
     struct InputModule {
-        IBaseModule::PTR module_ptr;
+        IBaseModule::WEAK_PTR module_ptr;
```

Comment:
As I known, shared_ptr just increase reference count, buffer will be kept, could you please provide more details about why it arise memory leak?